packages/net-stubbing/lib/server/intercepted-request.ts

```diff
@@ -184,7 +184,31 @@ export class InterceptedRequest {
           }
         }
 
-        if (eventName === 'before:request') {
+        if (
+          eventName === 'before:request' ||
+          // https://github.com/cypress-io/cypress/issues/17139
+          // This `cypressInternalResReplyCall` came
+          // from driver/src/cy/net-stubbing/events/before-requests.ts
+          // to sendStaticResponse at util.ts
+          // to middleware/request.ts
+          // to middleware/response.ts
+          // to here.
+          //
+          // This is a really ugly solution.
+          // But `cypressInternalResReplyCall` is necessary below because of the 3 reasons.
+          // 1. res.reply() call inside responseHandler inside cy.intercept()
+          // doesn't trigger `before:request` event.
```

Comment:
```suggestion
          // 1. req.reply() call inside responseHandler inside cy.intercept()
          // doesn't trigger `before:request` event.
```

also, what does this mean? `before:request` is what causes the request handler to be run, so this is not quite right